### PR TITLE
fix(AssetDrawer): do not close Drawer when interacting with notifications

### PR DESCRIPTION
**Pull request by SimonDegraeve**
## DESCRIPTION

For the `sourcing` project we have a Talent profile display in an `AssetDrawer` component, in this 
https://linear.app/wttj/issue/SOU-1064/z-index-with-growl-and-modal

## COMPATIBILITY

- [ ] Tested on Safari (desktop)
- [x] Tested on Chrome (desktop)
- [ ] Tested on Firefox (desktop)
- [ ] Tested on mobile device sizes
- [ ] Tested on tablet device sizes
- [ ] Tested on IOS Safari (either device or simulator)

## QA

- [x] Thoroughly tested in local environment
- [x] Added tests for all new features
- [ ] Added tests that considered edge cases

### CODE REVIEW

File: lib/src/components/Drawer/AssetDrawer/index.tsx [hunk 1]
```diff
@@ -15,12 +15,24 @@ export const AssetDrawerComponent = forwardRef<'div', AssetDrawerProps>(
...
+        element.contains(target)
+      )
+      return !isTargetWithinPersistentElements
+    }
+
     return (
       <Drawer
         {...rest}
         autoFocusOnShow={false}
+        getPersistentElements={getPersistentElements}
```
Comment (anucreative) [inline lib/src/components/Drawer/AssetDrawer/index.tsx:33]: We pass a `getPersistentElements` function already from `wttj-front`. Maybe we should put the `{...rest}` at the end of the props to allow overrides ?

**Review by anucreative — COMMENTED** [review]
Question about usage with other consumers

File: lib/src/components/Drawer/AssetDrawer/index.tsx (hunk 1 above)
Comment (SimonDegraeve) [reply lib/src/components/Drawer/AssetDrawer/index.tsx:48]: I updated the code so the behavior is the same as prior to this PR when you pass a `getPersistentElements()` prop

File: lib/src/components/Drawer/AssetDrawer/index.tsx [hunk 2]
```diff
@@ -12,15 +12,42 @@ import { CreateWuiProps, forwardRef } from '@/System'
...
+    const getPersistentElements = () =>
+      Array.from(
+        parentGetPersistentElements
+          ? parentGetPersistentElements()
+          : document.querySelectorAll('[data-wui-persistent]')
+      )
+
+    const hideOnInteractOutside =
+      Boolean(parentGetPersistentElements) ||
+      ((event: Event) => {
```
Comment (cnairi) [inline lib/src/components/Drawer/AssetDrawer/index.tsx:36]: Hi @SimonDegraeve 
I would extract the second condition in another variable for readability like:
```
const hideNotPersistentElement = ((event: Event) => {
      const target = event.target as HTMLElement
        const isTargetWithinPersistentElements = getPersistentElements().some(element =>
          element.contains(target)
        )
        return !isTargetWithinPersistentElements
}
```
(if possible)

Comment (SimonDegraeve) [reply lib/src/components/Drawer/AssetDrawer/index.tsx]: Done
